lemur/sources/service.py

```diff
@@ -89,6 +89,20 @@ def sync_endpoints(source):
 
         endpoint["certificate"] = certificate_service.get_by_name(certificate_name)
 
+        if not endpoint["certificate"] and certificate_name in certs_with_mismatched_name:
+            endpoint["certificate"] = certs_with_mismatched_name[certificate_name][0]
```

Comment:
Worth incrementing a metric of found by hash?

Reply:
added more metrics